django/forms/models.py

```diff
@@ -1158,8 +1159,9 @@ def __iter__(self):
         # Can't use iterator() when queryset uses prefetch_related()
         if not queryset._prefetch_related_lookups:
             queryset = queryset.iterator()
-        for obj in queryset:
-            yield self.choice(obj)
+
+        for distinct_choice in self.distinct_choices(queryset):
```

Comment:
Do we need a separate function, how about? 

```python 
        for obj in OrderedSet(queryset):
            yield self.choice(obj)
```

Reply:
I didn't know about the Django OrderedSet.
I've tried it and it works fine. I've made the update already. Thanks.

Reply:
This comment was about the need for `disticnt_choices()`. Do we need a function to create the `OrderedSet`. Can you create it right here like my code example? 🤔

Reply:
Oooh, I've gotten rid of the function.

Reply:
With ordered set, I get a sqlite3 failure... I think I'll have to revert to ordered dict